### PR TITLE
DAOS-7192 rebuild: Check if the container is destroyed

### PR DESCRIPTION
**Pull request by wangdi1**
If enumeration or fetch return -DER_NONEXIST, let's ignore
the failure if the container is destroyed.

For DER_BUSY, let's retry outside scan_request(), so it can
be delay for a few seconds.

Remove incorrect assertion for migrate tls.

Signed-off-by: Di Wang <di.wang@intel.com>

### CODE REVIEW

**Comment by wangdi1** [comment]
ping

File: src/object/srv_obj_migrate.c
Comment (liuxuezhao) [inline src/object/srv_obj_migrate.c:1321]: typo... destroyed

**Review by liuxuezhao — DISMISSED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by kprantis** [comment]
Looks like this patch is only waiting on its second review.

**Review by NiuYawei — APPROVED** [review]

**Comment by johannlombardi** [comment]
Please port to 1.2

**Pull request merged**
